### PR TITLE
use correct line endings per platform

### PR DESCRIPTION
**Pull request by Xaeroxe**
Fixes https://github.com/amethyst/tools/issues/49

### CODE REVIEW

**Review by torkleyy — COMMENTED** [review]
How could this happen? I never used Windows to edit the templates..

**Comment by Xaeroxe** [comment]
I believe it's because I used Windows to publish the tool on Cargo.  My git will automatically switch the line ending.

**Review by AnneKitsune — APPROVED** [review]
That could break templates at some point in the future, if someone makes a template with format!("blablabla\r\nblabla");

Its fine for now though, thanks!

**Comment by Rhuagh** [comment]
You can configure git to set a specific line ending type.

**Comment by Rhuagh** [comment]
https://help.github.com/articles/dealing-with-line-endings/

**Comment by Xaeroxe** [comment]
Sure, but then it'd be incorrect on Windows.  Regardless of how the templates are published this PR needs to switch the line endings as necessary.

File: src/new.rs [hunk 1]
```diff
@@ -30,7 +30,17 @@ impl New {
                 "__Cargo__.toml" => "Cargo.toml",
                 path => path,
             };
-            let content = content.replace("__project_name__", &self.project_name);
+            let mut content = content.replace("__project_name__", &self.project_name);
+            #[cfg(target_os = "windows")]
+            {
+                use regex::Regex;
+
+                content = Regex::new("[^\r]\n").unwrap().replace_all(&content, "\r\n").to_string();
```
Comment (AnneKitsune) [inline src/new.rs]: actually, change the regex to
[^\r]\n$

**Comment by Rhuagh** [comment]
Yeah, my point was just that by setting up .gitattributes correctly we can be sure that the template files will always be in a single format, which makes it easier to convert.

**Review by AnneKitsune — CHANGES_REQUESTED** [review]

**Comment by Xaeroxe** [comment]
@Rhuagh Even if we did do that it'd be easy to mess up if someone edited the templates again.  This way seems the easiest to maintain.

File: src/new.rs (hunk 1 above)
Comment (Xaeroxe) [reply src/new.rs]: I'm not sure that makes sense or is necessary.  \r\n is the end of the line, that's built into the definition.

Comment (AnneKitsune) [reply src/new.rs]: You can still format!("\r\n") and that would randomly get replaced by "\n" on linux, which can break templates. Same thing the other way around format!("line 1 \n line2").

Comment (AnneKitsune) [reply src/new.rs]: The $ just ensures that its the end of the line, and not somewhere in the middle of it.

Comment (Xaeroxe) [reply src/new.rs]: I think in order to match `format!("\r\n")` the regex would have to be \\\\r\\\\n.  This regex won't match that macro in source.

**Review by AnneKitsune — APPROVED** [review]

**Review by torkleyy — APPROVED** [review]

**Pull request merged**